u1qns/Week_09/BOJ_5002_도어맨.java

```diff
@@ -0,0 +1,44 @@
+import java.io.*;
+
+public class Main {
+	
+	static int getAnswer(char[] line, final int X)
+	{
+		int answer = 0;
+		int cnt[]  = new int[2];
+		int type = 0, tmp = 0;
+		
+		for (int i = 0; i < line.length; ++i) {
+			
+			type = (line[i] == 'W' ? 1 : 0);
+			tmp = (type == 1 ? 0 : 1);
+			
+			if(Math.abs(cnt[type] + 1 - cnt[tmp]) <= X)
+			{
+				++answer;
+				++cnt[type];
+			}
+			else if (i+1 < line.length && line[i+1] != line[i])
+			{
+				// 다음 사람이라도 입장 시킬 수 있는지 확인하는 로직
+				++answer;
+				++cnt[tmp];
```

Comment:
취향차이긴 한데, **tmp 변수** 대신 **비트 연산자**를 활용해보면 어떨까요?
서로 **상태가 다르다는 것**(=성별이 다르다는 것)을 좀 더 직관적으로 표현해줍니다!

```suggestion
			type = (line[i] == 'W' ? 1 : 0);
			
			if(Math.abs(cnt[type] + 1 - cnt[~type]) <= X)
			{
				++answer;
				++cnt[type];
			}
			else if (i+1 < line.length && line[i+1] != line[i])
			{
				// 다음 사람이라도 입장 시킬 수 있는지 확인하는 로직
				++answer;
				++cnt[~type];
```